### PR TITLE
Fill in GitHub App provider

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

This PR fills in the functions for the GitHub App provider delegate. This is the functionality that the provider will use once enrolled.

This PR does not add the functionality to enroll the GitHub App provider, or the functionality to generate an installation token credential.

Ref #2520

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: proto/minder/v1/minder.proto [hunk 1]
```diff
@@ -1232,6 +1232,18 @@ message GitHubProviderConfig {
...
+  string endpoint = 1;
+  // AppName is the GitHub App Name
+  string app_name = 2;
+  // AppID is the GitHub App ID
+  string app_id = 3;
+  // UserID is the GitHub App User ID
+  int64 user_id = 4;
+}
+
 // Context defines the context in which a rule is evaluated.
```
Comment (eleftherias) [inline proto/minder/v1/minder.proto:1247]: I'm unsure if all these fields belong in the provider definition. In the GitHub OAuth definition we only include the `endpoint`, but my assumption is if we're interacting with multiple GitHub endpoints (e.g public vs private), then each one will have it's own GitHub App.

Comment (JAORMX) [reply proto/minder/v1/minder.proto:1249]: So, for the OAuth2 app, we're able to point minder to an enterprise github endpoint. I'm not sure we can do the same with a github app flow? Is that a GitHub cloud-only kind of thing or is this something that may also work in on-prem installations? Would be good to figure out. Else, we can remove the `endpoint` for now and add it when we see the use-case.

**Review by JAORMX — COMMENTED** [review]
mostly lgtm, just got one comment.

File: proto/minder/v1/minder.proto (hunk 1 above)
Comment (eleftherias) [reply proto/minder/v1/minder.proto:1249]: GitHub Apps are available on GitHub Enterprise Server as well https://docs.github.com/en/apps/sharing-github-apps/making-your-github-app-available-for-github-enterprise-server#each-github-enterprise-server-instance-must-register-their-own-github-app

Comment (JAORMX) [reply proto/minder/v1/minder.proto:1249]: ack, let's include the endpoint then.

**Review by JAORMX — APPROVED** [review]

**Review by jhrozek — APPROVED** [review]

**Pull request merged**